tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp

```diff
@@ -119,7 +120,928 @@ void test_duration_output() {
+    fail_parse("dec 31 23:59:59 2019", "%c", ut);
+    fail_parse("dec 31 23:59:59 2019", "%c", st);
+    fail_parse("dec 31 23:59:59 2019", "%c", ft);
+
+    local_seconds lt;
+    test_parse("dec 31 23:59:59 2019", "%c", lt); // Not UTC, might be valid depending on the time zone.
+    assert(lt.time_since_epoch() == ref.time_since_epoch());
+
+    // Initially, TAI - UTC == 37s.
+    test_parse("dec 31 23:59:59 2019", "%c", tt);
+    test_parse("dec 31 23:59:22 2019", "%c", ut);
+    assert(tt == clock_cast<tai_clock>(ut));
+
+    test_parse("jan 01 00:00:35 2020", "%c", tt);
+    test_parse("dec 31 23:59:58 2019", "%c", ut);
+    assert(tt == clock_cast<tai_clock>(ut));
+
+    test_parse("jan 01 00:00:36 2020", "%c", tt);
+    test_parse("jan 01 00:00:00 2020", "%c", ut);
+    assert(tt == clock_cast<tai_clock>(ut));
+
+    // Initially, GPS - UTC == 18s
+    test_parse("dec 31 23:59:59 2019", "%c", gt);
+    test_parse("dec 31 23:59:41 2019", "%c", ut);
+    assert(gt == clock_cast<gps_clock>(ut));
+
+    test_parse("jan 01 00:00:16 2020", "%c", gt);
+    test_parse("dec 31 23:59:58 2019", "%c", ut);
+    assert(gt == clock_cast<gps_clock>(ut));
+
+    test_parse("jan 01 00:00:17 2020", "%c", gt);
+    test_parse("jan 01 00:00:00 2020", "%c", ut);
+    assert(gt == clock_cast<gps_clock>(ut));
+
+
+    ut_ref = utc_clock::from_sys(sys_days{1d / January / 2022y}) - 1s; // leap second insertion
+    test_parse("dec 31 23:59:60 2021", "%c", ut);
+    assert(ut == ut_ref);
+    test_parse("dec 31 23:59:60 2021", "%c", ft);
+    assert(ft == clock_cast<file_clock>(ut_ref));
+
+
+    // GH-1606: reads too many leading zeros
+    test_parse("19700405T000006", "%Y%m%dT%H%M%S", st);
+    assert(st == sys_days{5d / April / 1970y} + 6s);
+
+    // GH-1280 tests
+    year_month_day ymd;
+    test_parse("20200609", "%Y%m%d", ymd);
+    assert(ymd == 9d / June / 2020y);
+
+    test_parse("20201213", "%Y%m%d", ymd);
+    assert(ymd == 13d / December / 2020y);
+
+    test_parse("2020112", "%Y%m%d", ymd);
+    assert(ymd == 2d / November / 2020y);
+
+    test_parse("2020061125", "%Y%m%d", ymd);
+    assert(ymd == 11d / June / 2020y);
+
+    test_parse("2020120625119", "%Y%m%d", ymd);
+    assert(ymd == 6d / December / 2020y);
+
+    test_parse("2020092Text", "%Y%m%d", ymd);
+    assert(ymd == 2d / September / 2020y);
+
+    test_parse("20200609", "%Y%m%d", ymd);
+    assert(ymd == 9d / June / 2020y);
+
+    test_parse("2020112", "%Y%m%d", ymd);
+    assert(ymd == 2d / November / 2020y);
+
+    test_parse("2020061125", "%Y%m%d", ymd);
+    assert(ymd == 11d / June / 2020y);
+
+    test_parse("2020124", "%Y%m%d", ymd);
+    assert(ymd == 4d / December / 2020y);
+
+    test_parse("2020104Text", "%Y%m%d", ymd);
+    assert(ymd == 4d / October / 2020y);
+
+    fail_parse("202000000000000923", "%Y%m%d", ymd);
+    fail_parse("202000000000000923", "%Y%m%d", ymd);
+}
+
+void test_parse() {

```

Comment:
:warning: Open issue: We should have some test coverage for non-`char` types. Doesn't need to be extensive, just minimal verification that `wchar_t` compiles and appears to behave reasonably for a few scenarios.